R/load_runner.R

```diff
@@ -15,14 +15,18 @@
 #' @param dry_run If TRUE then any database changes are rolled back when the
 #' import completes. i.e. the load stage can be run and tests executed but the
 #' db will be rolled back.
+#' @param comment An optional comment to add to the import log table for this
+#' run.
 #'
 #' @keywords internal
 #'
 run_load <- function(con, load, transformed_data, test_queries, path,
-                     test_file, dry_run) {
+                     test_file, dry_run, log_table, comment) {
   if (is.null(transformed_data)) {
     stop("Cannot run tests as no data has been transformed.")
   }
+  log_data <- get_log_data(path, comment)
+  verify_log_table(con, log_table, log_data)
```

Comment:
It will be worth checking here, just before load, that the import has not yet been run.

(Even if you add the name as a PK it would be nice to get a human-understandable error message here)

Reply:
Added a check now and test that message is human readable.